adapter/src/main/kotlin/com/sun/tools/jdi/KDACommandLineLauncher.kt

```diff
@@ -0,0 +1,298 @@
+package com.sun.tools.jdi
+
+import com.sun.jdi.Bootstrap
+import com.sun.jdi.VirtualMachine
+import com.sun.jdi.connect.Connector
+import com.sun.jdi.connect.IllegalConnectorArgumentsException
+import com.sun.jdi.connect.VMStartException
+import com.sun.jdi.connect.spi.Connection
+import com.sun.jdi.connect.spi.TransportService
+import java.io.File
+import java.io.IOException
+import java.io.InterruptedIOException
+import java.net.URLDecoder
+import java.net.URLEncoder
+import java.nio.charset.StandardCharsets
+import java.util.*
+
+internal const val ARG_HOME = "home"
+internal const val ARG_OPTIONS = "options"
+internal const val ARG_MAIN = "main"
+internal const val ARG_INIT_SUSPEND = "suspend"
+internal const val ARG_QUOTE = "quote"
+internal const val ARG_VM_EXEC = "vmexec"
+internal const val ARG_CWD = "cwd"
+internal const val ARG_ENVS = "envs"
+
+
+class KDACommandLineLauncher : SunCommandLineLauncher {
+
+    companion object {
+
+        fun urlEncode(arg: Collection<String>?) = arg
+                ?.map { URLEncoder.encode(it, StandardCharsets.UTF_8.name()) }
+                ?.fold("") { a, b -> "$a\n$b" }
+
+        fun urlDecode(arg: String?) = arg
+                ?.trim('\n')
+                ?.split("\n")
+                ?.map { URLDecoder.decode(it, StandardCharsets.UTF_8.name()) }
+                ?.toList()
+    }
+
+    constructor() : super() {
+        addStringArgument(
+                ARG_CWD,
+                ARG_CWD,
+                "Current working directory",
+                "",
+                false)
+        addStringArgument(
+                ARG_ENVS,
+                ARG_ENVS,
+                "Environment variables",
+                "",
+                false)
+    }
+
+    override fun name(): String? {
+        return this.javaClass.name
+    }
+
+    override fun description(): String? {
+        return "A custom launcher supporting cwd and env variables"
+    }
+
+    /**
+     * Copied from SunCommandLineLauncher.java and added cwd / env processing logic
```

Comment:
Thanks for the contribution. I am a bit unsure of whether this might be a licensing issue. The Kotlin debug adapter is licensed very permissively whereas JDK implementations might not be. At the very least we may have to retain their license headers.

Reply:
Thanks for the advice!  I'll have a deeper investigation to this and fix the license issue.